examples/_Tutorials/dataframes_intro.ipynb

```diff
@@ -0,0 +1,761 @@
+    "\n",
+    "### Unsupported DataFrames\n",
+    "\n",
+    "Unlike Pandas, BodoDataFrames cannot support arbitrary Python types in columns. Each column in a BodoDataFrame should have a single supported type. Supported types include ints, floats, bool, decimal128, timestamps/datetime, dates, durations/timedelta, string, binary, list, map, and struct. \n",
+    "\n",
+    "Some examples of unsupported DataFrames:\n",
+    "\n",
+    "``` py\n",
+    "# Mixed-type tuples are not supported (use structs instead)\n",
+    "DataFrame({\"A\": [(\"a\", 1), (\"b\", 2)]})\n",
+    "\n",
+    "# Each column must hold a single type\n",
+    "DataFrame({\"A\": [1, \"a\"]})\n",
+    "\n",
+    "# DataFrames cannot have arbitrary Python objects\n",
+    "DataFrame({\"A\": [MyObject()] * 4})\n",
+    "```\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Lazy Evaluation and Plans\n",
+    "\n",
+    "In Bodo, operations on DataFrames and Series are lazy, meaning that they return a lazy result representing a DataFrame, Series or Scalar which contains some metadata i.e. a schema, but not the actual data itself. Instead, lazy results contain a \"plan\" attribute. A plan is an expression tree describing how to go from data sources to the current object using relational operators like join, aggregate, or project.\n",
+    "\n",
+    "Lazy evaluation allows Bodo to optimize the plan before execution, which can have a huge impact (e.g. 100x) on workload performance. Common optimizations include reordering joins, pushing filters to I/O, or eliminating dead columns."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To see an example of lazy evaluation, we will create a DataFrame, representing a Parquet read over a billion row dataset (NYC taxi). Normally, this dataset would be too large to fit into memory on most laptops, however since the `read_parquet` API is lazy, no actual data is materialized. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas\n",
+    "import bodo.pandas as pd\n",
+    "from bodo.ext import plan_optimizer\n",
+    "\n",
+    "df = pd.read_parquet(\"s3://bodo-example-data/nyc-taxi/fhvhv_tripdata\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can immediately inspect the metadata of our lazy DataFrame such as the column names and data types. When `read_parquet` is called, Bodo opens the first couple parquet files in the dataset to infer the schema, which is typically pretty fast. \n",
+    "\n",
+    "We can also look at the plan for this DataFrame. Bodo uses DuckDB plans as an intermediary representation to perform optimizations using the DuckDB optimizer. \n",
```

Comment:
Mentioning DuckDB seems confusing to me in general. We can rename `generate_duckdb` or just leave it there for now but avoid mentioning DuckDB in the text because it sounds like DuckDB is doing the computation.